File: app/src/main/java/com/kieran/winnipegbusbackend/ServiceAdvisories/ServiceAdvisory.java
package com.kieran.winnipegbusbackend.ServiceAdvisories;

import com.kieran.winnipegbusbackend.StopTime;

import java.io.Serializable;
import java.util.List;

public class ServiceAdvisory implements Serializable, Comparable {
    private String title;
    private String header;
    private List<AffectedStop> affectedStops;
    private List<Reroute> reroutes;
    private StopTime updatedAt;

    public ServiceAdvisory(String title, String header, List<AffectedStop> affectedStops, List<Reroute> reroutes, StopTime updatedAt) {
        this.title = title;
        this.header = header;
        this.affectedStops = affectedStops;
        this.reroutes = reroutes;
        this.updatedAt = updatedAt;
    }

    public String getTitle() {
        return title;
    }

    public List<Reroute> getReroutes() {
        return reroutes;
    }

    public StopTime getUpdatedAt() {
        return updatedAt;
    }

    public String getHeader() {
        return header;
    }

    public List<AffectedStop> getAffectedStops() {
        return affectedStops;
    }

    @Override
    public int compareTo(Object another) {
        return -updatedAt.compareTo(((ServiceAdvisory)another).getUpdatedAt());
    }
}

File: app/src/main/java/com/kieran/winnipegbusbackend/winnipegtransit/ServiceAdvisories/ServiceAdvisory.kt
package com.kieran.winnipegbusbackend.winnipegtransit.ServiceAdvisories

import com.kieran.winnipegbusbackend.common.StopTime

import java.io.Serializable
import java.util.ArrayList

class ServiceAdvisory(val title: String, val header: String, val affectedStops: List<AffectedStop>, val reroutes: List<Reroute>, val updatedAt: StopTime) : Serializable, Comparable<Any> {

    val affectedStopsStrings: List<String>
        get() {
            val strings = ArrayList<String>()

            for (stop in affectedStops) {
                strings.add(stop.affectedStop)
            }

            return strings
        }

    override operator fun compareTo(another: Any): Int {
        return -updatedAt.compareTo((another as ServiceAdvisory).updatedAt)
    }
}
